app/ordered_map.go

```diff
@@ -33,7 +33,7 @@ func NewOrderedMapFromPairs[K comparable, V any](pairs []Pair[K, V]) *OrderedMap
 	newMap := NewOrderedMap[K, V]()
 
 	for _, pair := range pairs {
-		newMap.Set(pair.Key, pair.Value)
+		newMap.SetNew(pair.Key, pair.Value)
```

Comment:
This ensures exclusivity of the `pair.Key` in the input list = that input list do NOT contain more then single pair with the same `pair.Key` value.